src/vmm/src/devices/virtio/mmio.rs

```diff
@@ -186,18 +186,21 @@ impl MmioTransport {
             DRIVER_OK if self.device_status == (ACKNOWLEDGE | DRIVER | FEATURES_OK) => {
                 self.device_status = status;
                 let device_activated = self.locked_device().is_activated();
-                if !device_activated
-                    && self.are_queues_valid()
-                    && self.locked_device().activate(self.mem.clone()).is_err()
-                {
-                    self.device_status |= DEVICE_NEEDS_RESET;
-
-                    // Section 2.1.2 of the specification states that we need to send a device
-                    // configuration change interrupt
-                    let _ = self
-                        .locked_device()
-                        .interrupt_trigger()
-                        .trigger_irq(IrqType::Config);
+                if !device_activated && self.are_queues_valid() {
+                    // temporary variable needed for borrow checker
+                    let activate_result = self.locked_device().activate(self.mem.clone());
+                    if let Err(err) = activate_result {
+                        self.device_status |= DEVICE_NEEDS_RESET;
+
+                        // Section 2.1.2 of the specification states that we need to send a device
+                        // configuration change interrupt
+                        let _ = self
+                            .locked_device()
+                            .interrupt_trigger()
+                            .trigger_irq(IrqType::Config);
+
+                        log::error!("Failed to activate virtio device: {}", err)
```

Comment:
nit: just import `log::error` (or `logger::error`, I think we should use this everywhere for consistency)

Reply:
done